cl/corpus_importer/management/commands/harvard_opinions.py

```diff
@@ -566,57 +635,102 @@ def clean_docket_number(docket_number: str) -> str:
     return docket_number
 
 
-def clean_body_content(case_body: str) -> str:
+def clean_body_content(case_body: str, harvard: bool = False) -> str:
     """Strip all non-alphanumeric characters
 
     :param case_body: Opinion text
+    :param harvard: Are we harvard xml data
     :return:Opinion text with only alphanumeric characters
     """
     soup = BeautifulSoup(case_body, "lxml")
-    return re.sub(r"[^a-zA-Z0-9]", "", soup.text)
+    if not harvard:
+        opinion_text = soup.text
+    else:
+        opinions = []
+        for op in soup.find_all("opinion"):
+            opinions.append(op.text)
+        opinion_text = "".join([op.text for op in soup.find_all("opinion")])
+
+    return re.sub(r"[^a-zA-Z0-9]", "", opinion_text.lower())
 
 
 def match_based_text(
```

Comment:
This function actually has some pretty nice filtering that it does, but it's easy to miss the big picture because it's so long. I think it'd be much neater if this function was something like:

```
for case in possible_cases:
    if any(
        length_too_different(some, args),
        content_too_different(some, args),
        case_names_dont_overlap(some, args),
        case_names_too_different(some, args),
        cosine_similarity_too_low(some, args),
    ):
        continue
```
I know that's a somewhat sizeable refactor, but having written this kind of code three times myself, I find that you're on the cusp of merging code that's almost readable. This last piece would — I think — take it over the line, so you could glance at it and know what kind of filtering it was doing.

Reply:
great.  Sorry I missed this until now.